### PR TITLE
Support parsing guarding Pokemon display

### LINKED ISSUE #164: Parse PokemonFortProto.GuardPokemonDisplay
In `PokemonFortProto` there is `GuardPokemonDisplay` which we do not parse at the moment. This causes the frontend to fail to display correct form/costume of the guarding Pokemon.

Related to that, there is also `GuardPokemonLevel` which could be useful?

### PR DESCRIPTION
**Pull request by Mygod**
Fixes #164. Needs testing.

### CODE REVIEW

File: sql/27_guard_display.up.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE `gym`
+    ADD COLUMN `guarding_pokemon_display` text DEFAULT NULL;
```
Comment (Fabio1988) [inline sql/27_guard_display.up.sql:2]: After guarding_pokemon_id pls

**Comment by Fabio1988** [comment]
Hmm do we really need full display? I guess much better than adding more separate columns 😂

Wonder if this is really sent in the GMO proto...

**Comment by Mygod** [comment]
It is sent because in the overworld you can see these (forms, badges, shiny, etc). It's not separate columns since I don't think anyone would ever want to filter on these data.

File: decoder/gym.go [hunk 2]
```diff
@@ -156,11 +158,32 @@ func calculatePowerUpPoints(fortData *pogo.PokemonFortProto) (null.Int, null.Int
...
 	gym.GuardingPokemonId = null.IntFrom(int64(fortData.GuardPokemonId))
+	display, _ := json.Marshal(pokemonDisplay{
+		Form:          int(fortData.GuardPokemonDisplay.Form),
+		Costume:       int(fortData.GuardPokemonDisplay.Costume),
+		Gender:        int(fortData.GuardPokemonDisplay.Gender),
+		Shiny:         fortData.GuardPokemonDisplay.Shiny,
+		TempEvolution: int(fortData.GuardPokemonDisplay.CurrentTempEvolution),
+		Alignment:     int(fortData.GuardPokemonDisplay.Alignment),
+		Badge:         int(fortData.GuardPokemonDisplay.PokemonBadge),
+		LocationCard:  int(fortData.GuardPokemonDisplay.LocationCard.LocationCard),
```
Comment (Fabio1988) [inline decoder/gym.go:184]: This causes `SIGSEGV: segmentation violation`... as it tries to access `LocationCard` which is not set

**Review by Fabio1988 — CHANGES_REQUESTED** [review]

**Comment by Fabio1988** [comment]
Pls fix the Segmentation violation, would have merged this. tested but it doesn't work

File: decoder/gym.go (hunk 2 above)
Comment (Mygod) [reply decoder/gym.go]: Fixed ty.

Comment (Fabio1988) [reply decoder/gym.go]: Is the fallback to 0 on purpose? What's the value though if available?

Comment (Fabio1988) [reply decoder/gym.go]: I wonder if we also need to nil check the display in general, empty gyms? Can't test right now

Comment (Mygod) [reply decoder/gym.go]: Hmm good point. `0` means unset so it is on purpose.

**Comment by Fabio1988** [comment]
@Mygod did you check this with trafficlight or any other proto display tool?
I am running this now for 3 hours, the column is still empty

**Comment by Fabio1988** [comment]
fixed your lazy PR ;) 

```
{"form":946,"costume":0,"gender":1,"shiny":false,"temp_evolution":0,"alignment":0,"badge":0,"location_card":0}
```

**Review by Fabio1988 — APPROVED** [review]
seems to work in my testing environment.. so what's the plan on that? will RM support the data?

**Comment by Mygod** [comment]
Yes... RM will support everything. :)

**Comment by Fabio1988** [comment]
> Yes... RM will support everything. :)

without maintainer? :cry:

**Comment by lenisko** [comment]
sounds like @Mygod is willing to maintain it as well!

**Comment by Mygod** [comment]
@TurtIeSocks Please help support this. 🥺

**Pull request merged**